mmengine/visualization/vis_backend.py

```diff
@@ -610,3 +610,135 @@ def close(self):
         """close an opened tensorboard object."""
         if hasattr(self, '_tensorboard'):
             self._tensorboard.close()
+
+
+@VISBACKENDS.register_module()
+class MLFlowVisBackend(BaseVisBackend):
+    """MLFlow visualization backend class.
+
+    It can write images, config, scalars, etc. to a
+    mlflow file.
+
+    Examples:
+        >>> from mmengine.visualization import MLFlowVisBackend
+        >>> import numpy as np
+        >>> vis_backend = MLFlowVisBackend(save_dir='temp_dir')
+        >>> img = np.random.randint(0, 256, size=(10, 10, 3))
+        >>> vis_backend.add_image('img', img)
+        >>> vis_backend.add_scaler('mAP', 0.6)
+        >>> vis_backend.add_scalars({'loss': 0.1,'acc':0.8})
+        >>> cfg = Config(dict(a=1, b=dict(b1=[0, 1])))
+        >>> vis_backend.add_config(cfg)
+
+    Args:
+        save_dir (str): The root directory to save the files
+            produced by the backend.
+        exp_name (str, optional): The experiment name. Default to None.
+        tags (dict, optional): The tags to be added to the experiment.
+            Default to None.
+        params (dict, optional): The params to be added to the experiment.
+            Default to None.
+        log_model (bool, optional): Whether to log the model. Default to True.
+    """
+
+    def __init__(self,
+                 save_dir: str,
+                 exp_name: Optional[str] = None,
+                 tags: Optional[dict] = None,
+                 params: Optional[dict] = None,
```

Comment:
I'm not very familiar with MLFlow, but  it seems that the params arguments passed to `mlflow.log_params` is more like the config file parsed by `Runner`. How about using `mlflow.log_params` to record the config file?

Reply:
https://mlflow.org/docs/latest/python_api/mlflow.html?highlight=log_params#mlflow.log_params
Looking at the MLFlow documentation, log_prams appears to be a function that stores the parameters used for actual training. What do you think about writing a ```config.py``` that applies to that learning?

tools\train.py ```config.py```

Reply:
I think the `config.py` store the main parameters for training. If MLFlow provides an interface to save the configuration, we should call it to save it.

Reply:
Success! I will clean up the code and commit soon. 😊
![image](https://user-images.githubusercontent.com/66167254/212718785-ead9d9b8-fdf8-4f64-9652-8a5a34a0de2d.png)